### DISCUSSION TITLE
[Bug]: nginx port stuck on 80

### DISCUSSION THREAD

**Discussion by kolaval** [beyondcode/herd-community#761] [General]
### Platform

Windows

### Operating system version

Windows 11 Pro

### System architecture

Windows

### Herd Version

1.6.1

### PHP Version

8.3 (8.3.7)

### Bug description

I updated the %USERPROFILE%\.config\herd\config\nginx\herd.conf with a new port:
listen 127.0.0.1:8050 default_server;

Nginx service still fails to start with : bind() to 127.0.0.1:80 failed (10013: An attempt was made to access a socket in a way forbidden by its access permissions)

### Steps to reproduce

Change nginx port in I updated the %USERPROFILE%\.config\herd\config\nginx\herd.conf
Start nginx service

**Comment by mpociot** [comment]
Do you have any secured/isolated sites? Those could have individual configuration files, that still use port 80.

While you can manually modify the port in your config files, Herd is currently not designed for this and some parts might break by changing the port.

**Reply by powlam** [reply]
I'm in a similar situation: my company has blocked the port 80 of localhost, so I need to change the port for Nginx manually at `herd.conf`.
It would be great to **allow changing the port of Nginx from Herd's UI**; that way it would be under controll and Herd would guarantee that nothing breaks ;)

**Comment by patrickodacre** [comment]
@mpociot Are you able to provide more detail on what would break if we change this port manually, please? While it would be nice to have a simple option in the app itself, if there were a documented workaround things wouldn't be so bad. From comments like this - https://github.com/beyondcode/herd-community/issues/860#issuecomment-2308023786 - you can see that Windows 11 users have a less-than-straightforward way to deal with this. It isn't great that we should disable core features of Windows 11 just to get Herd to work.

So to clarify, my first request would be for some details on what would "break" should we change the nginx port in all the config files.
My second request would be for some documentation on the subject, because piecing together all these details from GitHub issues was a bit of a hassle.

**Comment by ecomit** [comment]
I also have Win11, edited herd.conf. nginx still does not start, because of the same error. It is a blank install. No sites. 
Error [emerg] 5412#26568: bind() to 127.0.0.1:80 failed (10013: An attempt was made to access a socket in a way forbidden by its access permissions)

**Reply by Aryazaky** [reply]
It seems this bug is not planned to be fixed for windows users, reading from #1366 , #860 , and #848

**Comment by itxshakil** [comment]
In my case:

### 🚨 Problem

Nginx couldn’t start on port 80:

```
bind() to 127.0.0.1:80 failed (10013: An attempt was made to access a socket in a way forbidden by its access permissions)
```

---

### 🔎 Investigation

1. Ran:

   ```cmd
   netsh http show servicestate
   ```
2. The output showed:

   ```
   Registered URLs:
       HTTP://*:80/
       Request queue name: DefaultAppPool
   ```

   ✅ This confirmed that **IIS (DefaultAppPool)** was reserving port 80 via `http.sys`.

---

### ✅ Solution

If IIS isn’t needed, disable it to free up port 80:

```cmd
sc stop W3SVC
sc config W3SVC start= disabled
```

Then restart Nginx — it will bind to port 80 successfully.

---

### 🔑 Recommendation

Since `netsh http show servicestate` showed **DefaultAppPool**, the cleanest fix is to **disable IIS** if you only need Nginx.
